src/Build.UnitTests/Graph/ProjectGraph_Tests.cs

```diff
@@ -37,6 +38,256 @@ public void TestGraphWithSingleNode()
+                TransientTestProjectWithFiles projectA =
+                    env.CreateTestProjectWithFiles(projectAContents, Array.Empty<string>(), "..\\A");
+                TransientTestProjectWithFiles projectB =
+                    env.CreateTestProjectWithFiles(projectBContents, Array.Empty<string>(), "..\\B");
+
+                List<string> projectsToParse = new List<string>();
+                projectsToParse.Add(projectA.ProjectFile);
+                ProjectGraph graph = new ProjectGraph(projectsToParse);
+                graph.ProjectNodes.Count.ShouldBe(2);
+            }
+        }
+
+        /// <summary>
+        /// A  B
+        /// \ /
+        ///  C
+        /// </summary>
+        [Fact]
+        public void TestGraphWithMultipleEntryPointsMultipleNodes()
+        {
+            string projectAContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+                  <PropertyGroup>
+                    <TestProperty>value</TestProperty>
+                  </PropertyGroup>
+                  <ItemGroup>
+                    <ProjectReference Include=""..\C\build.proj"" />
+                  </ItemGroup>
+                </Project>
+                ";
+
+            string projectBContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+                  <PropertyGroup>
+                    <TestProperty>value</TestProperty>
+                  </PropertyGroup>
+                  <ItemGroup>
+                    <ProjectReference Include=""..\C\build.proj"" />
+                  </ItemGroup>
+                </Project>
+                ";
+
+            string projectCContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+                  <PropertyGroup>
+                    <TestProperty>value</TestProperty>
+                  </PropertyGroup>
+                </Project>
+                ";
+
+            using (var env = TestEnvironment.Create())
+            {
+                TransientTestProjectWithFiles projectA =
+                    env.CreateTestProjectWithFiles(projectAContents, Array.Empty<string>(), "..\\A");
+                TransientTestProjectWithFiles projectB =
+                    env.CreateTestProjectWithFiles(projectBContents, Array.Empty<string>(), "..\\B");
+                TransientTestProjectWithFiles projectC =
+                    env.CreateTestProjectWithFiles(projectCContents, Array.Empty<string>(), "..\\C");
+                List<string> projectsToParse = new List<string>();
+                projectsToParse.Add(projectA.ProjectFile);
+                projectsToParse.Add(projectB.ProjectFile);
+                ProjectGraph graph = new ProjectGraph(projectsToParse);
+
+                graph.ProjectNodes.Count.ShouldBe(3);
+                ProjectGraphNode projectNodeA = graph.ProjectNodes.First(node => node.Project.FullPath.Equals(projectA.ProjectFile));
```

Comment:
We'll need to think about exactly what the final public graph API is at some point. Should nodes have a "refereng**ing**" field (if we could ever come up with a good name there)? Or a lookup by `Project`/`ProjectInstance`/project file name?

This may become clear as we try to start actually using it and also when integrating this in with QuickBuild.